java_tests/src/test/java/org/lsst/sal/SAL_TestTest.java

```diff
@@ -0,0 +1,427 @@
+package org.lsst.sal;
+
+import Test.*;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class SAL_TestTest extends BaseTestCase {
+
+    @Test
+    public void testGetCurrentTime() {
+        salUtils su = new salUtils();
+        // Run the test twice in the unlikely event that the first run
+        // occurs at a leap second transition.
+        int leapSeconds = 0;
+        double measuredSeconds = 0.0;
+        for (int i = 0; i < 2; i++) {
+            leapSeconds = su.getLeapSeconds();
+            measuredSeconds = remote.getCurrentTime() - (System.currentTimeMillis() / 1000.0);
+            if (Math.abs(leapSeconds - measuredSeconds) - 1 < 0.1) {
+                // We are probably at a leap second transition;
+                // sleep and try again.
+                sleep(2);
+                continue;
+            }
+            break;
+        }
+        Assert.assertEquals(leapSeconds, measuredSeconds, 0.001);
+    }
+
+    @Test
+    @Ignore("No astropy equivalent for java.")
+    public void testGetLeapSeconds() {
+    }
+
+    @Test
+    public void testGetVersions() {
+        final String salVersion = remote.getSALVersion();
+        Assert.assertNotNull(salVersion);
+        Assert.assertNotSame("", salVersion);
+        final String xmlVersion = remote.getXMLVersion();
+        Assert.assertNotNull(xmlVersion);
+        Assert.assertNotSame("", xmlVersion);
+    }
+
+    /**
+     * Write several logevent messages and make sure gettting the oldest returns the data in the expected order.
+     *
+     * @throws Exception In case of a problem related to reflection.
```

Comment:
What does `reflection` mean in this context? Please expand this comment (and the many others like it).

Reply:
Reflection is a Java standard present in the JDK and JVM for introspection of classes and class instances. Much like getattr in Python. Any Java developer knows what reflection means in a Java context so from a Java point of view there is no need to explain this.